addons/source-python/packages/source-python/messages/hooks.py

```diff
@@ -182,7 +182,11 @@ def _pre_send_user_message(args):
         _recipients.update(*tuple(tmp_recipients), clear=True)
         args[1] = _recipients
 
-        buffer = make_object(ProtobufMessage, args[3])
+        try:
+            buffer = make_object(ProtobufMessage, args[3])
+        except RuntimeError:
+            # Patch for issue #390 - UserMessage was created by another plugin.
+            buffer = ProtobufMessage.from_abstract_pointer(args[3])
```

Comment:
The problem though, is that any changes made to the `buffer`'s fields within a plugin's callback will have no effect because `args[3]` is never mutated. We could simply overwrite it with our own, but that is likely to cause some scoping issues since unlike bitbuf messages, I believe protobuf ones can be nested, etc. I went ahead and added `ProtobufMessage.parse_to_abstract_pointer` so that we can ensure any changes that were made is properly applied to the original buffer. For example, something like this (untested):

```python
        # No need to do anything behind this if no listener is registered
        if not user_message_hooks and not protobuf_user_message_hooks:
            return

        try:
            # Replace original recipients filter
            tmp_recipients = make_object(BaseRecipientFilter, args[1])
        except RuntimeError:
            # Patch for issue #314
            tmp_recipients = RecipientFilter.from_abstract_pointer(args[1])
        _recipients.update(*tuple(tmp_recipients), clear=True)
        args[1] = _recipients

        try:
            buffer = make_object(ProtobufMessage, args[3])
        except RuntimeError:
            # Patch for issue #390 - UserMessage was created by another plugin.
            buffer = ProtobufMessage.from_abstract_pointer(args[3])

        protobuf_user_message_hooks.notify(_recipients, buffer)

        # No need to do anything behind this if no listener is registered
        if user_message_hooks:
            try:
                impl = get_user_message_impl(message_index)
            except NotImplementedError:
                impl = None

            if impl is not None:
                data = impl.read(buffer)
                user_message_hooks.notify(_recipients, data)

                # Update buffer if data has been changed
                if data.has_been_changed():
                    buffer.clear()
                    impl.write(buffer, data)

        # If we wrapped the buffer from an abstract pointer, make sure to
        #   apply any changes that may have been made back into the original.
        if get_object_pointer(buffer) != args[3]:
            buffer.parse_to_abstract_pointer(args[3])
```

If this works for you, please update this PR and adjust if needs be. Thanks!